lib/travis/logs/services/fetch_log.rb

```diff
@@ -0,0 +1,41 @@
+module Travis
+  module Logs
+    module Services
+      class FetchLog
+        def initialize(database: nil)
+          @database = database || Travis::Logs.database_connection
+        end
+
+        attr_reader :database
+        private :database
+
+        def run(job_id: nil, id: nil)
+          return nil if job_id.nil? && id.nil?
+          if job_id && id
+            raise ArgumentError, 'only one of job_id or id allowed'
+          end
+
+          result = nil
```

Comment:
In this case, I would like to keep it for clarity.  I personally dislike Ruby's rules about nil initialization when using dangling conditions.